helm/hpcc/templates/eclscheduler.yaml

```diff
@@ -42,6 +42,11 @@ data:
 {{ include "hpcc.generateGlobalConfigMap" .root | indent 6 }}
 {{- end -}}
 
+{{- if empty $.Values.dali -}}
+  {{- if $.Values.eclscheduler -}}
+     {{- $_ := fail (printf "eclscheduler requires missing DALI" ) -}}
```

Comment:
instead of performing this check for each case,
could you pass context (for the key name and name) and check inside/when daliArg is called?

i.e. I assume if a template gets as far as calling daliArg, it needs it.